controllers/pravegacluster_controller.go

```diff
@@ -434,6 +434,20 @@ func (r *PravegaClusterReconciler) reconcileSegmentStoreService(p *pravegav1beta
 				return fmt.Errorf("failed to update headless service port (%s): %v", currentService.Name, err)
 			}
 		}
+		if len(currentService.Spec.Ports) == 1 {
+			currentService.Spec.Ports = append(currentService.Spec.Ports, headlessService.Spec.Ports[1])
+			err = r.Client.Update(context.TODO(), currentService)
```

Comment:
I'd suggest to collect all changes to the service and run the `Update()` command only once after Getting it (there is another potential Update call on line 433).